### PR TITLE
Building boost with the updated buildenv

### PR DESCRIPTION
**Pull request by pspoerri**
See PR here: https://github.com/C2SM-RCM/buildenv/pull/5/files

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,9 @@ endif ()
 # Python installation path 
 set (PYTHON_PATH "python")
 
+# Add boost include dir to includes:
+INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
```
Comment (andyspiros) [inline CMakeLists.txt]: We use lower case for cmake functions. But it's not a big deal...

**Comment by andyspiros** [comment]
Looks good.

**Comment by pspoerri** [comment]
Ok, I made the include directories lower case.

**Comment by pspoerri** [comment]
launch jenkins

**Comment by pspoerri** [comment]
Based on the test results from jenkins I added a couple of checks and adapted the buildenv again: https://github.com/C2SM-RCM/buildenv/pull/6/files

**Comment by pspoerri** [comment]
launch jenkins

**Comment by pspoerri** [comment]
launch jenkins

File: CMakeLists.txt [hunk 2]
```diff
@@ -23,9 +23,16 @@ set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/fortran )
 
 install(DIRECTORY ${CMAKE_BINARY_DIR}/fortran/ DESTINATION include/fortran FILES_MATCHING PATTERN "*.mod" PATTERN "CMake*" EXCLUDE )
 
-set(CXX11_FLAGS "-std=c++11")
-
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+# Based on http://stackoverflow.com/a/31010221/592024
+if (CMAKE_VERSION VERSION_LESS "3.1")
+    if("${CMAKE_Fortran_COMPILER}" MATCHES "ftn")
```
Comment (andyspiros) [inline CMakeLists.txt]: Wait a second, are you sure this works on CSCS machines? `ftn` is the fortran compiler wrapper, so if you have the `PrgEnv-gnu` loaded `ftn` will represent the GNU compiler.

I think you need something more sophisticated to take the decision here.

File: test/build.sh [hunk 3]
```diff
@@ -16,8 +16,9 @@ cmakeConfigure()
     local CMAKEARGS=(..
            "-DCMAKE_Fortran_COMPILER=${fcomp}"
            "-DCMAKE_INSTALL_PREFIX=${idir}"
-           "-DCMAKE_CXX_COMPILER=g++"
-           "-DCMAKE_C_COMPILER=gcc"
+           "-DCMAKE_CXX_COMPILER=CC"
+           "-DCMAKE_C_COMPILER=cc"
```
Comment (andyspiros) [inline test/build.sh]: Isn't it a better idea to set the environmental variables `CC` (for C) and `CXX` (for C++). This is what I always did and used to be the recommended way.

Comment (clementval) [inline test/build.sh]: +1

Comment (andyspiros) [inline test/build.sh]: *3

Comment (andyspiros) [inline test/build.sh]: sqrt(.)

Comment (pspoerri) [inline test/build.sh]: This change depends on https://github.com/C2SM-RCM/buildenv/pull/7 I will adapt it accordingly.

Comment (andyspiros) [inline test/build.sh]: As far as I understand, the compiler should always be `cc` for C and `CC` for C++ on Cray machines. The wrapper takes care of compiling with the right compiler and link against the correct version of the MPI libraries and other runtimes.

**Comment by pspoerri** [comment]
launch jenkins

File: test/build.sh (hunk 3 above)
Comment (clementval) [inline test/build.sh]: So is there a configuration problem on kesch ? because if you don't load PrgEnv-cray, you don't have any `cc` or `CC`

Comment (andyspiros) [inline test/build.sh]: I don't know, actually. I have only experience with the other Cray machines, not with Kesch. How do you compile with MPI in Kesch?

**Comment by pspoerri** [comment]
launch jenkins

File: test/build.sh (hunk 3 above)
Comment (pspoerri) [inline test/build.sh]: CC/cc/gcc/g++ are wrapped in such a way that you get MPI when the correct modules are loaded. We now solve the problem directly within the buildenv where we simply expose `CXX`, `CC` and `FC`. CMake will pick these environment variables up and do whatever it wants with it.

**Comment by pspoerri** [comment]
Once jenkins has finished. I will merge it and install the configuration.

**Pull request merged**